loopchain/blockchain/vote.py

```diff
@@ -69,39 +66,38 @@ def votes(self):
     def target_hash(self):
         return self.__target_hash
 
-    @staticmethod
-    def __make_vote_init(audience):
+    def __make_vote_init(self, audience):
         if not audience:
             return None
 
         vote_init = {}
         if isinstance(audience, PeerManager):
-            for group_id in list(audience.peer_list.keys()):
-                if group_id == conf.ALL_GROUP_ID:
-                    continue
-                vote_init[group_id] = {}
-                for peer_id in list(audience.peer_list[group_id].keys()):
-                    vote_init[group_id][peer_id] = []
-        else:
-            for peer_id in audience:
-                vote_init[audience[peer_id].group_id] = {}
-                vote_init[audience[peer_id].group_id][peer_id] = []
+            audience = list(audience.peer_list[conf.ALL_GROUP_ID].keys())
+
+        for peer_id in audience:
+            vote_init[peer_id] = None
 
         logging.debug("vote_init: " + str(vote_init))
         return vote_init
 
-    @staticmethod
-    def __parse_vote_sign(vote_sign):
-        """서명된 vote로 부터 투표 결과를 추출한다."""
+    def __parse_vote_sign(self, vote_sign):
+        """Derive result of vote from vote_sign."""
 
         return vote_sign
 
-    def add_vote(self, group_id, peer_id, vote_sign):
-        if group_id not in self.__votes.keys():
+    def add_vote(self, peer_id, vote_sign):
+        if peer_id not in self.__votes.keys():
             return False
-        if peer_id not in self.__votes[group_id].keys():
+
+        result = self.__parse_vote_sign(vote_sign)
+
+        if self.__votes[peer_id]:
```

Comment:
This can be applied `early return`.
Can you place it upper line?

Then `else` of this if statement can be deleted.

Reply:
👌